src/components/Card/ItemCard/index.tsx

```diff
@@ -1,33 +1,53 @@
-import { ReactNode } from 'react'
+import { ReactElement, ReactNode } from 'react'
 import { ItemCardPanel, ItemContentPanel, ItemDetailPanel } from './styled'
 
-export interface ItemCardData {
+export interface ItemCardData<T> {
   title: string
-  content: ReactNode
+  render: (data: T, index: number) => ReactNode
 }
 
-export const ItemDetail = ({ item, hideLine }: { item: ItemCardData; hideLine: boolean }) => (
-  <ItemDetailPanel hideLine={hideLine}>
-    <div className="item__detail__title">{item.title}</div>
-    <div className="item__detail__value">{item.content}</div>
-  </ItemDetailPanel>
-)
-
-export default ({
+export function ItemCard<T>({
   items,
+  data,
   children,
   outputIndex,
 }: {
-  items: ItemCardData[]
+  items: ItemCardData<T>[]
+  data: T
   children?: ReactNode
   outputIndex?: string
-}) => (
-  <ItemCardPanel id={outputIndex ? `output_${outputIndex}` : ''}>
-    <ItemContentPanel>
-      {items.map((item, index) => (
-        <ItemDetail key={item.title} item={item} hideLine={index === items.length - 1} />
+}): ReactElement {
+  return (
+    <ItemCardPanel id={outputIndex ? `output_${outputIndex}` : ''}>
```

Comment:
`outputIndex === 0` leads to an empty `id`, is it by design? I see it's intended to be `undefined` in some cases(https://github.com/nervosnetwork/ckb-explorer-frontend/pull/1174/files#diff-324f9ebd00c9d399ad683152356ab34e6d21ed711ab0f39b90c35989f96b6fd6R41)

Reply:
> is it by design?

This is code that existed before the refactoring, which I guess is not designed, and I can fix it by the way in this PR.

Reply:
I checked the code and found that the outputIndex property has not been passed anywhere, it may be useless for now, consider removing it directly in this refactoring?

Reply:
It’s fine to remove it in this PR